nncf/common/quantization/structs.py

```diff
@@ -0,0 +1,175 @@
+from collections import namedtuple
+from copy import deepcopy
+from enum import Enum
+from typing import Dict, List, Optional
+
+
+class QuantizationMode:
+    SYMMETRIC = "symmetric"
+    ASYMMETRIC = "asymmetric"
+
+
+class QuantizerConfig:
+    """A generic, framework-agnostic information on a configuration of a quantizer for abstract reasoning
+    and determination of a quantizer setup scheme for a given model."""
+    def __init__(self, num_bits: int = 8,
+                 mode: QuantizationMode = QuantizationMode.SYMMETRIC,
+                 signedness_to_force: Optional[bool] = None,
+                 per_channel: bool = False):
+        self.num_bits = num_bits
+        self.mode = mode
+        self.signedness_to_force = signedness_to_force
+        self.per_channel = per_channel
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+    def __str__(self):
+        return "B:{bits} M:{mode} SGN:{signedness} PC:{per_channel}".format(
+            bits=self.num_bits,
+            mode='S' if self.mode == QuantizationMode.SYMMETRIC else 'A',
+            signedness='ANY' if self.signedness_to_force is None else ('S' if self.signedness_to_force else 'U'),
+            per_channel='Y' if self.per_channel else 'N')
+
+    def __hash__(self):
+        return hash(str(self))
+
+    def is_valid_requantization_for(self, other: 'QuantizerConfig') -> bool:
+        #pylint:disable=too-many-return-statements
+        """Quantizer config A is a valid requantization for quantizer config B if A is more strict -
+        specifically, it might be reasonable to put quantizer A after quantizer B in tensor data control flow, so that
+        the requantization will further constrain the input tensor data w.r.t. values it can take, but
+        putting quantizer A after quantizer B would be unreasonable."""
+        if self.num_bits > other.num_bits:
+            return False
+        if self.mode is QuantizationMode.ASYMMETRIC and other.mode is \
+                QuantizationMode.SYMMETRIC:
+            return False
+        if self.signedness_to_force is None and other.signedness_to_force is not None:
+            return False
+        if self.signedness_to_force is True and other.signedness_to_force is False:
+            return False
```

Comment:
```suggestion
        conditions = [
            self.num_bits > other.num_bits,
            self.mode is QuantizationMode.ASYMMETRIC and other.mode is QuantizationMode.SYMMETRIC,
            self.signedness_to_force is None and other.signedness_to_force is not None,
            self.signedness_to_force is True and other.signedness_to_force is False
        ]
        if any(conditions):
            return False
```

You can also write comments for all non-trivial conditions.

Reply:
This kind of code does not allow early exits, but I guess Python isn't about performance anyway. Done.